ui/hooks/reservation.tsx

```diff
@@ -81,31 +111,57 @@ export const useOrder = (
       onError: () => {},
     });
 
+  const { data, error, loading } = useQuery<Query, QueryReservationByPkArgs>(
+    GET_RESERVATION,
+    {
+      fetchPolicy: "no-cache",
+      variables: { pk: reservationPk },
+      skip: !reservationPk,
+    }
+  );
+
   return {
-    order: data,
-    error: error != null,
-    refreshError,
-    loading: orderLoading || refreshLoading,
-    refresh,
+    reservation: data?.reservationByPk,
+    error,
+    loading,
     deleteReservation,
     deleteError,
     deleteLoading,
-    called,
     deleted,
   };
 };
 
-export const useReservation = (
-  reservationPk: number
-): { reservation: ReservationType; error: ApolloError; loading: boolean } => {
-  const { data, error, loading } = useQuery<Query, QueryReservationByPkArgs>(
-    GET_RESERVATION,
+type UseReservationsProps = {
+  currentUser: UserType;
+  states?: ReservationsReservationStateChoices[];
+  orderBy?: string;
+};
+
+export const useReservations = ({
+  currentUser,
+  states,
+  orderBy,
+}: UseReservationsProps): {
+  reservations: ReservationType[];
+  error: ApolloError;
+  loading: boolean;
+} => {
+  const { data, error, loading } = useQuery<Query, QueryReservationsArgs>(
+    LIST_RESERVATIONS,
     {
+      skip: !currentUser?.pk,
+      variables: {
+        ...(states?.length > 0 && { state: states }),
+        ...(orderBy && { orderBy }),
+        user: currentUser?.pk.toString(),
```

Comment:
Number.toString() is unsafe. undefined.toString() throws unlike String() constructor.